### ISSUE TITLE
Unable to select keyboard on iOS (Safari)

### ISSUE THREAD

**Issue by RaspberryLime** [freeCodeCamp/freeCodeCamp#43233]
## Affected page

https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/say-hello-to-html-elements

## To reproduce

1. Go to https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/say-hello-to-html-elements
2. Click on the “Code” tab.
3. Look at the bottom right of the screen. The button to bring up the keyboard is almost completely hidden. It is exceedingly difficult to click it.

## Expected behavior

I’d like to see the button and be able to reliably click it. 

## Screenshots

![D6866661-A91E-45C0-AE7B-986944487090](https://user-images.githubusercontent.com/25120737/129916849-dd7e8895-11ca-4c24-a3dd-2944cbb006c9.png)

## System

- Device: iPhone 8.
- OS: iOS 14.7.1.
- Browser: Safari.
- Version: ?

**Comment by ieahleen** [comment]
Does it appear if you scroll up?
![03436560-1526-46F1-A63E-05B32B46F1C5](https://user-images.githubusercontent.com/26656284/129919453-3536ee02-7cc7-4370-89c5-230b4132c721.jpeg)

**Comment by RaspberryLime** [comment]
> Does it appear if you scroll up?

I was going to say that it doesn’t scroll up, but if I try a dozen or more times to scroll/swipe up it might hit lucky and actually scroll up.

**Comment by ieahleen** [comment]
That doesn't seem ideal at all, the buttons below should be accessible. It seems the issue is bigger than first reported

**Comment by RaspberryLime** [comment]
Also, if you have to scroll down to see the button (if or when scrolling works), you can’t see what or where you’re typing either. Very occasionally it’s possible when I can hit the sliver of button without scrolling, the coding section seems to work really well, so it’s a shame this is spoiling it.

**Comment by ieahleen** [comment]
I'll add the Help wanted label, someone may be able to figure out how to make the buttons accessible on mobile.

An issue with the _help wanted_ or _first timers only_ label is open for contribution. The first comprehensive PR created will be reviewed and merged. We typically do not assign issues to anyone other than long-time contributors.

If you would like to contribute, and have not read the contributors docs, please do so here: https://contribute.freecodecamp.org/#/

If you have any issues with contributing, be sure to join us on the [contributors channel](https://chat.freecodecamp.org/channel/contributors), or on the [contributors sub-forum](https://forum.freecodecamp.org/c/contributors/3)

**Comment by moT01** [comment]
Scrolling doesn't work when you try to scroll the editor. You need to try to scroll the whole web page by swiping the nav menu or the buttons at the bottom.

**Comment by Nirajn2311** [comment]
The scrolling seems to be weird in iPhone. Its like there is one main scroller which controls the main view between navbar and the bottom bar and another scroller which controls the challenge description.

**Comment by moT01** [comment]
Perhaps a little off issue - but one thing we could maybe do is add some PWA to the website. That would remove the browser buttons and address bar if you add it to your home screen. Giving more space.

If you go to coderadio.freecodecamp.org/ on your phone and add it to the home screen - it has this. Open it from your home screen - the browser stuff is gone.

The main site doesn't have it - so the browser stuff is there if you add it to the home screen. One downside is that there's no browser back button, so if you go somewhere in the same tab where you can't get back, your stuck and would probably have to close and reopen the site. I think anyway.

**Comment by RaspberryLime** [comment]
This is fixed in iOS 15. There are two ways of displaying tabs in Safari now, and it works both ways. 

This is the new (default) way of displaying tabs and the URI bar in Safari:
![6E2D39E3-C1AC-4F94-8060-0DE285418D8F](https://user-images.githubusercontent.com/25120737/134816711-2e0fce97-c812-4cb3-adfc-4272f9a02202.png)

It is possible to switch to the old way:
![2C333507-BE47-40B2-AB5B-276329038709](https://user-images.githubusercontent.com/25120737/134816713-c4b26301-d530-4c25-be79-0e048dc40701.png)

**Comment by ieahleen** [comment]
This seems fixed, so I am closing the issue. If I am wrong please comment to have it reopened.

**Comment by philross49** [comment]
> This seems fixed, so I am closing the issue. If I am wrong please comment to have it reopened.

Still problems with iOS and Safari on mobile!

Have to zoom out, then in to make the keyboard button appear. Can’t scroll up or down when in the code tab. Tilting for landscape brings up the keyboard button, but reduces usable screen to nothing. Rotating back the portrait gets rid of the keyboard and the button.

iOS mobile keyboard is also not great for this, I’ll be continuing on my MacBook!

![74A8ABE9-911A-4C67-A87C-C60C927CFAEF](https://user-images.githubusercontent.com/94573114/142291298-5962bf0f-2b47-4cb1-bcf2-d49f33e8f4ad.png)
![A214F4DD-1936-48D4-9CDC-648987D06D6A](https://user-images.githubusercontent.com/94573114/142291306-084dde4c-ad1c-403c-9b18-f317162ac989.png)

**Comment by RaspberryLime** [comment]
> > This seems fixed, so I am closing the issue. If I am wrong please comment to have it reopened.
> 
> Still problems with iOS and Safari on mobile!
> 
> Have to zoom out, then in to make the keyboard button appear. Can’t scroll up or down when in the code tab. Tilting for landscape brings up the keyboard button, but reduces usable screen to nothing. Rotating back the portrait gets rid of the keyboard and the button.
> 
> iOS mobile keyboard is also not great for this, I’ll be continuing on my MacBook!
> 
> ![74A8ABE9-911A-4C67-A87C-C60C927CFAEF](https://user-images.githubusercontent.com/94573114/142291298-5962bf0f-2b47-4cb1-bcf2-d49f33e8f4ad.png) ![A214F4DD-1936-48D4-9CDC-648987D06D6A](https://user-images.githubusercontent.com/94573114/142291306-084dde4c-ad1c-403c-9b18-f317162ac989.png)

Do you have a link to the specific pages with the issue?

I’m finding some still work OK for me whilst others have changed. The very first page has changed like you describe for me. 
![B2B9FA58-6471-4735-B24B-3B5C26411A87](https://user-images.githubusercontent.com/25120737/142403332-f233013d-b78c-4352-9ea5-851a83746bc9.png)

**Comment by apvarghese19** [comment]
I will attempt to fix this issue

**Comment by ahmadabdolsaheb** [comment]
@moT01, let me know if you need help with this.

**Comment by moT01** [comment]
I glanced at this a little - I didn't have a way to reproduce it on my computer, so I wasn't able to open the browser dev tools to play around with it and find a fix. So basically, I would have been kind of just guessing on a fix, and then checking my phone to see if it worked. I could look around a little harder for some tools that would help me.

**Comment by bbsmooth** [comment]
As soon as PR [#48586](https://github.com/freeCodeCamp/freeCodeCamp/pull/48586) is merged we should verify if this is still an issue.

**Comment by Sembauke** [comment]
![IMG_81EAB29D7086-1](https://github.com/freeCodeCamp/freeCodeCamp/assets/46919888/efe42d78-6cd5-4d00-aacf-26c8c5959582)

This seems to be fixed. Please re-open if there are still issues.
